plugin/core/main.py

```diff
@@ -1,29 +1,132 @@
 import sublime
+import sublime_plugin
 
 from ..color import remove_color_boxes
 from ..diagnostics import DiagnosticsPresenter
 from ..highlights import remove_highlights
+from .handlers import LanguageHandler
 from .logging import set_debug_logging, set_exception_logging
 from .panels import destroy_output_panels, ensure_panel, PanelName
 from .popups import popups
-from .registry import windows, load_handlers, unload_sessions
+from .protocol import Response
+from .protocol import WorkspaceFolder
+from .registry import windows, unload_sessions
+from .rpc import method2attr
+from .sessions import AbstractPlugin
+from .sessions import register_plugin
+from .sessions import Session
+from .settings import client_configs
 from .settings import settings, load_settings, unload_settings
-from .typing import Optional
+from .transports import kill_all_subprocesses
+from .types import ClientConfig
+from .typing import Optional, List, Type, Callable, Dict
+import weakref
 
 
 def ensure_server_panel(window: sublime.Window) -> Optional[sublime.View]:
     return ensure_panel(window, PanelName.LanguageServers, "", "", "Packages/LSP/Syntaxes/ServerLog.sublime-syntax")
 
 
+def _get_final_subclasses(derived: List[Type], results: List[Type]) -> None:
+    """
+    This function should be removed: https://github.com/sublimelsp/LSP/issues/899
+    """
+    for d in derived:
+        d_subclasses = d.__subclasses__()
+        if len(d_subclasses) > 0:
+            _get_final_subclasses(d_subclasses, results)
+        else:
+            results.append(d)
+
+
+def _forcefully_register_plugins() -> None:
+    """
+    This function should be removed: https://github.com/sublimelsp/LSP/issues/899
+    """
+    plugin_classes = []  # type: List[Type[AbstractPlugin]]
+    _get_final_subclasses(AbstractPlugin.__subclasses__(), plugin_classes)
+    for plugin_class in plugin_classes:
+        register_plugin(plugin_class)
+    language_handler_classes = []  # type: List[Type[LanguageHandler]]
+    _get_final_subclasses(LanguageHandler.__subclasses__(), language_handler_classes)
+    for language_handler_class in language_handler_classes:
+        # Create an ephemeral plugin that stores an instance of the LanguageHandler as a class instance. Custom requests
+        # and notifications will work.
+        class LanguageHandlerTransition(AbstractPlugin):
+
+            handler = language_handler_class()
+            has_regular_name = False
+
+            @classmethod
+            def name(cls) -> str:
+                n = cls.handler.name  # type: ignore
+                if n.lower().startswith('lsp-'):
+                    cls.has_regular_name = True
+                    return n[len('lsp-'):]
+                return n
```

Comment:
That part breaks backward-compatibility with existing project-wide overrides. It will be expected to use `typescript` now, instead of `lsp-typescript`, for example.

It might be acceptable but since all the other changes strive to make everything backward-compatible, it would be shame to have that one thing break.